nucypher/cli/actions/select.py

```diff
@@ -160,10 +164,17 @@ def select_config_file(emitter: StdoutEmitter,
 
             parsed_config_files.append(fp)
             parsed_addresses_and_filenames.append([config_checksum_address, Path(fp).name])  # store checksum & filename
-        except config_class.OldVersion:
-            # no use causing entire usage to crash if file can't be used anyway - inform the user; they can
-            # decide for themself
-            emitter.echo(IGNORE_OLD_CONFIGURATION.format(config_file=fp), color='yellow')
+        except config_class.OldVersion as e:
+            # attempt migration
```

Comment:
Note: this implies that any config file encountered with an older version is migrated - even if we are only reading the file eg. `nucypher ursula config` to show the contents of the file.  Instead of failing and telling the user to migrate the configuration using the `migrate` function, we automatically migrate the file, and then can continue to show the contents of the file.

Just double-checking if we are comfortable with this.

Reply:
If someone is running the `config` CLI commands it's safe to assume it's interactive-safe.  It's totally fine to prompt here if you choose to.

Reply:
Yeah this is the tricky part indeed. We can do this. We would need an extra parameter like `do_auto_migrate` for `select_config_file` which defaults to `False`.

Then, any reading calls don't do the migration, and then only set `do_auto_migrate` to True, when Ursula run is making the call. We don't want to prompt during `nucypher ursula run`.

Reply:
✔️ - resolved. User is prompted by default unless `do_auto_migrate` is set to True. It is currently only set to try for `nucypher ursula run`.